gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckClassUniquenessTask.java

```diff
@@ -0,0 +1,120 @@
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+@SuppressWarnings("checkstyle:designforextension") // making this 'final' breaks gradle
+public class CheckClassUniquenessTask extends DefaultTask {
+
+    private Configuration configuration;
+
+    public CheckClassUniquenessTask() {
+        setGroup("Verification");
+        setDescription("Checks that the given configuration contains no identically named classes.");
+    }
+
+    @Input
+    public Configuration getConfiguration() {
+        return configuration;
+    }
+
+    public void setConfiguration(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    @TaskAction
+    public void checkForDuplicateClasses() {
+        ClassUniquenessAnalyzer analyzer = new ClassUniquenessAnalyzer(getLogger());
+        analyzer.analyzeConfiguration(getConfiguration());
+        boolean success = analyzer.getProblemJars().isEmpty();
+        writeResultFile(success);
+
+        if (!success) {
+            analyzer.getProblemJars().forEach((problemJars) -> {
+                Set<String> classes = analyzer.getDuplicateClassesInProblemJars(problemJars);
+                getLogger().error("Identically named classes found in {} jars ({}): {}",
+                        problemJars.size(), problemJars, classes);
+            });
+
+            throw new IllegalStateException(String.format(
+                    "'%s' contains multiple copies of identically named classes - "
+                            + "this may cause different runtime behaviour depending on classpath ordering.\n"
+                            + "To resolve this, try excluding one of the following jars, "
+                            + "changing a version or shadowing:\n\n%s",
+                    configuration.getName(),
+                    formatSummary(analyzer)
+            ));
+        }
+    }
+
+    private static String formatSummary(ClassUniquenessAnalyzer summary) {
+        int maxLength = summary.jarsToClasses().keySet().stream().flatMap(Set::stream)
+                .map(ModuleVersionIdentifier::toString)
+                .map(String::length)
+                .max(Comparator.naturalOrder()).get();
+        String format = "%-" + (maxLength + 1) + "s";
+
+        Map<String, Integer> sortedTable = summary.jarsToClasses().entrySet().stream().collect(Collectors.toMap(
+                entry -> entry.getKey().stream().map(jar -> String.format(format, jar)).collect(Collectors.joining()),
+                entry -> entry.getValue().size(),
+                (first, second) -> {
+                    throw new RuntimeException("Unexpected collision: " + first + ", " + second);
+                },
+                TreeMap::new));
+
+        StringBuilder builder = new StringBuilder();
+        sortedTable.forEach((jars, classes) ->
+                builder.append(String.format("\t%-14s", "(" + classes + " classes) ") + jars + "\n"));
```

Comment:
why not include "(" and ")" within the string.format template?

Reply:
The `%-14s` handles right-padding with spaces, so I want to supply the fully parenthesised numbers as input and have the padding appended after the last paren.